tools/package.js

```diff
@@ -19,6 +22,46 @@ function main() {
 
 main();
 
+function getPublishedVersion() {
```

Comment:
I suggest to rollback all the changes, related to versioning - it is a complex question (you can just always bump the patch version), and anyway, it is A LOT of tools we could adopt instead (if we really want), for example [cocogitto](https://github.com/cocogitto/cocogitto).

And that's very fragile to pick the latest version from npmjs.com: better to use annotated tags Git history, for example (because not every version should be published).